common/utils/history/groupByPeriodUnit.js

```diff
@@ -91,23 +92,40 @@ function computeMissionGroups(missions, periodProps) {
   return groupsByPeriodUnit;
 }
 
-export function useGroupMissionsAndExtractActivities(missions, periodProps) {
+export function useGroupMissionsAndExtractActivities(
+  missions,
+  start,
+  end,
+  periodProps
+) {
   const [activities, setActivities] = React.useState([]);
 
   const [
     missionGroupsByPeriodUnit,
     setMissionGroupsByPeriodUnit
   ] = React.useState({});
 
+  const missionInPeriod = (mission, startTime, endTime) =>
+    mission.startTime < endTime &&
+    (!mission.endTime || mission.endTime > startTime);
   React.useEffect(() => {
-    setMissionGroupsByPeriodUnit(computeMissionGroups(missions, periodProps));
-    const acts = missions.reduce(
+    const from = startOfMonth(start);
+    const to = isThisMonth(end) ? end : endOfMonthAsDate(end);
+    const fromTime = jsToUnixTimestamp(from.getTime());
+    const toTime = jsToUnixTimestamp(to.getTime());
+    const filteredMissions = missions.filter(mission =>
+      missionInPeriod(mission, fromTime, toTime)
+    );
```

Comment:
When retrieving missions from period filter, we add it to the store, so the input missions to be grouped by period unit (day, week, etc.) contains them all. We had to filter them according the period filter before grouping by period unit.